kevm-pyk/src/kevm_pyk/kproj/evm-semantics/evm.md

```diff
@@ -1728,21 +1727,37 @@ Precompiled Contracts
     rule #precompiled(8) => ECPAIRING
     rule #precompiled(9) => BLAKE2F
 
-    syntax Set ::= #precompiledAccounts ( Schedule ) [klabel(#precompiledAccounts), function, total]
+    syntax Int ::= #precompiledAccounts ( Schedule ) [klabel(#precompiledAccounts), function, total]
  // ------------------------------------------------------------------------------------------------
-    rule #precompiledAccounts(DEFAULT)           => SetItem(1) SetItem(2) SetItem(3) SetItem(4)
+    rule #precompiledAccounts(DEFAULT)           => 4
     rule #precompiledAccounts(FRONTIER)          => #precompiledAccounts(DEFAULT)
     rule #precompiledAccounts(HOMESTEAD)         => #precompiledAccounts(FRONTIER)
     rule #precompiledAccounts(TANGERINE_WHISTLE) => #precompiledAccounts(HOMESTEAD)
     rule #precompiledAccounts(SPURIOUS_DRAGON)   => #precompiledAccounts(TANGERINE_WHISTLE)
-    rule #precompiledAccounts(BYZANTIUM)         => #precompiledAccounts(SPURIOUS_DRAGON) SetItem(5) SetItem(6) SetItem(7) SetItem(8)
+    rule #precompiledAccounts(BYZANTIUM)         => 8
     rule #precompiledAccounts(CONSTANTINOPLE)    => #precompiledAccounts(BYZANTIUM)
     rule #precompiledAccounts(PETERSBURG)        => #precompiledAccounts(CONSTANTINOPLE)
-    rule #precompiledAccounts(ISTANBUL)          => #precompiledAccounts(PETERSBURG) SetItem(9)
+    rule #precompiledAccounts(ISTANBUL)          => 9
     rule #precompiledAccounts(BERLIN)            => #precompiledAccounts(ISTANBUL)
     rule #precompiledAccounts(LONDON)            => #precompiledAccounts(BERLIN)
     rule #precompiledAccounts(MERGE)             => #precompiledAccounts(LONDON)
     rule #precompiledAccounts(SHANGHAI)          => #precompiledAccounts(MERGE)
+
+    syntax Set ::= #precompiledAccountsAsSet ( Schedule ) [klabel(#precompiledAccountsAsSet), function, total]
```

Comment:
Another implementation that wouldn't require any maintance:

```
rule #precompiledAccountsAsSet(SCHED) => #precompiledAccountsAsSetAux(#precompiledAccounts(SCHED))
rule #precompiledAccountsAsSetAux(N) => .Set requires N <Int 0
rule #precompiledAccountsAsSetAux(N) => SetItem(N) #precompiledAccountsAsSetAux(N -Int 1) [owise]
```

Since we'll always pass in a concrete `N`, should be fine.

Reply:
Yes, I was just thinking of doing something like that, that's the way to go.